client/components/domains/trademark-claims-notice/index.jsx

```diff
@@ -0,0 +1,298 @@
+import { getSelectedSite } from 'state/ui/selectors';
+import TrademarkNotice from './trademark-notice';
+
+/**
+ * Style dependencies
+ */
+import './style.scss';
+
+class TrademarkClaimsNotice extends React.Component {
+	static propTypes = {
+		domain: PropTypes.string,
+		isSignupStep: PropTypes.bool,
+		onAccept: PropTypes.func,
+		onGoBack: PropTypes.func,
+		onReject: PropTypes.func,
+		recordAcknowledgeTrademarkButtonClickInTrademarkNotice: PropTypes.func,
+		recordChooseAnotherDomainButtonClickInTrademarkNotice: PropTypes.func,
+		recordShowTrademarkNoticeButtonClickInTrademarkNotice: PropTypes.func,
+		trademarkClaimsNoticeInfo: PropTypes.object,
+	};
+
+	static defaultProps = {
+		analyticsSection: 'domains',
+		trademarkClaimsNoticeInfo: null,
+		isSignupStep: false,
+	};
+
+	state = this.getDefaultState();
+
+	getDefaultState() {
+		return {
+			hasScrolledToBottom: false,
+			showFullNotice: false,
+			trademarkClaimsNoticeInfo: this.props.trademarkClaimsNoticeInfo,
+			finishedFetching: ! isEmpty( this.props.trademarkClaimsNoticeInfo ),
+		};
+	}
+
+	UNSAFE_componentWillMount() {
+		if ( isEmpty( this.props.trademarkClaimsNoticeInfo ) && ! this.state.finishedFetching ) {
+			this.checkDomainAvailability().then( ( { trademarkClaimsNoticeInfo } ) => {
+				this.setState( {
+					trademarkClaimsNoticeInfo,
+					finishedFetching: true,
+				} );
+			} );
+		}
+	}
+
+	componentWillUnmount() {
+		window.removeEventListener( 'scroll', this.handleScroll );
+	}
+
+	checkDomainAvailability = () => {
+		const { domain } = this.props;
+
+		return new Promise( resolve => {
+			checkDomainAvailability(
+				{
+					domainName: domain,
+					blogId: get( this.props, 'selectedSite.ID', null ),
+					isCartPreCheck: false,
+				},
+				( error, result ) => {
+					resolve( {
+						trademarkClaimsNoticeInfo: get( result, 'trademark_claims_notice_info', null ),
+					} );
+				}
+			);
+		} );
+	};
+
+	enableActionButtons = () => {
+		this.setState( { hasScrolledToBottom: true } );
+		window.removeEventListener( 'scroll', this.handleScroll );
+	};
+
+	handleScroll = () => {
+		const { hasScrolledToBottom } = this.state;
+		if ( hasScrolledToBottom ) {
+			return;
+		}
+
+		const element = document.scrollingElement;
+
+		if ( element.scrollHeight - element.scrollTop < element.clientHeight + 100 ) {
+			this.enableActionButtons();
+		}
+	};
+
+	renderHeader = () => {
+		const { onGoBack, domain, translate } = this.props;
+
+		return (
+			<HeaderCake onClick={ onGoBack }>
+				{ translate( 'Register %(domain)s', { args: { domain } } ) }
+			</HeaderCake>
+		);
+	};
+
+	renderPreamble = () => {
+		const { domain, translate } = this.props;
+
+		return (
+			<CompactCard>
+				<h2>{ translate( '%(domain)s matches a trademark.', { args: { domain } } ) }</h2>
+				<p>
+					{ translate(
+						"To continue, you must agree not to infringe on the trademark holders's rights. Please review and acknowledge the following notice."
```

Comment:
`holders's` => `holders'`?

Reply:
🤦‍♂ This is what I get for copy/pasting without reviewing closely.  Thanks!